### PR TITLE
Tweaks to root url

### PR DESCRIPTION
**Pull request by imbstack**
This does two things. First one we've agreed upon. Second I'm happy to remove if we don't want it.

1. Change `TASKCLUSTER_ROOT` to `TASKCLUSTER_ROOT_URL`
2. Uses taskcluster-lib-urls to build urls

### CODE REVIEW

**Comment by djmitche** [comment]
Per https://docs.google.com/document/d/1B_pvuY4X4R05Rj0-FTrDZMTGQqqId76IIHMcrCAFcW0/edit we need to be able to generate /docs and / and /tools URLS from the rootUrl, too, so it shouldn't have /api in it.  Right?

**Comment by jhford** [comment]
It feels like forcing part of the value to be `/api` is maybe not something we want to enforce at the client instantiation level.

**Comment by imbstack** [comment]
I also felt a bit weird about forcing `api/` but we're going to be doing it at some point anyway. We could allow people to set their own path prefix for services when they set up the load balancer I suppose, but that ends up being more trouble than it is worth, right?

We have `api/` in the path currently so it will need to come from somewhere. It has to be either in the name of the service (instead of being called `auth`, called `api/auth`), or in the `rootUrl` or come from somewhere else. I think the plan in [bug 1459261](https://bugzilla.mozilla.org/show_bug.cgi?id=1459261) is to use that library to build a url for each endpoint based on a `rootUrl` that does not contain `api/` but also still end up forcing the value to be `api/`. Is that ok with you, @jhford?

File: README.md [hunk 1]
```diff
@@ -670,13 +670,13 @@ var auth = new taskcluster.Auth({
...
 different instances of taskcluster. This can also be set by setting a
-`TASKCLUSTER_ROOT` env var before importing taskcluster-client. You can also
+`TASKCLUSTER_ROOT_URL` env var before importing taskcluster-client. You can also
 use global config options as below:
 
 ```js
 // Configure default options
 taskcluster.config({
-  rootUrl: "https://somesite.com",
+  rootUrl: "https://somesite.com/api",
```
Comment (djmitche) [inline README.md]: ..meaning i think this should stay `https://somesite.com`.

File: src/client.js [hunk 2]
```diff
@@ -155,22 +155,19 @@ exports.createClient = function(reference, name) {
...
-      baseUrl:          reference.baseUrl        || '',
       exchangePrefix:   reference.exchangePrefix || '',
+      // We can remove the second half of this assignment once all api definitions are upgraded to have `name`
+      serviceName: reference.name || reference.baseUrl.split('//')[1].split('.')[0],
+      serviceVersion: 'v' + (reference.version + 1),
     }, _defaultOptions);
 
-    // Remove possible trailing slash from baseUrl
-    this._options.baseUrl = this._options.baseUrl.replace(/\/$/, '');
+    assert(this._options.rootUrl, 'Must provide a rootUrl or set the env var TASKCLUSTER_ROOT_URL');
```
Comment (djmitche) [inline src/client.js]: Maybe worth having a `tcUrl.isValidRootUrl(..)` to check this, and clean its trailing slash..

Comment (imbstack) [reply src/client.js]: The only problem is that I think we're pretty liberal with what a valid root url is, so I'm not sure how much we could validate really.

Comment (djmitche) [reply src/client.js]: Well we can be as conservative as we want :)

I was thinking
 * has schema and it's http or https
 * has a hostname
 * has no path, or has path / (we should enforce one of those two)

**Pull request merged**